lib/supavisor/db_handler.ex

```diff
@@ -50,26 +50,28 @@ defmodule Supavisor.DbHandler do
   def handle_event(:internal, _, :connect, %{auth: auth} = data) do
     Logger.info("Try to connect to DB")
 
-    socket_opts = [
+    sock_opts = [
       :binary,
       {:packet, :raw},
-      {:active, true},
+      {:active, false},
       auth.ip_version
     ]
 
-    case :gen_tcp.connect(auth.host, auth.port, socket_opts) do
-      {:ok, socket} ->
+    case :gen_tcp.connect(auth.host, auth.port, sock_opts) do
+      {:ok, sock} ->
         Logger.debug("auth #{inspect(auth, pretty: true)}")
+        sock = {:gen_tcp, sock}
 
-        msg =
-          :pgo_protocol.encode_startup_message([
-            {"user", auth.user},
-            {"database", auth.database},
-            {"application_name", auth.application_name}
-          ])
+        case handshake(sock, auth) do
+          {:ok, sock} ->
+            # Back to active from here
+            set_sock_opts(sock, [{:active, true}])
```

Comment:
nitpick: I'd consider calling this `setopts` to match socket API, I think it will be immediately obvious. You can also use shorthand `active: true`:

```suggestion
            setopts(sock, active: true)
```

btw, since you only ever use it for socket activation you can instead do:

```
activate(sock)
```

I vaguely recall we have exactly that little helper in Postgrex.

It is totally fine as is of course!

Reply:
that looks much better 👍